### PR TITLE
implemented routes, protected routes, and redirect to login page if no JWT token in localstorage

### PR DESCRIPTION
**Pull request by Pradip-Singh-X**

### CODE REVIEW

**Review by sudip-sth — COMMENTED** [review]
- Make title short and add the relevant info in description
- Repo name should be in small case `dashboard-practice`

File: src/context/AuthContext.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+export const AuthProvider: React.FC<{ children: React.ReactNode }> = ({
+  children,
+}) => {
+  const [user, setUser] = useState<UserType | null>(null);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const initAuth = async () => {
+      const authToken = localStorage.getItem("authToken");
```
Comment (sudip-sth) [inline src/context/AuthContext.tsx]: make the const for localStorage keys as it must be accessed across multiple components

File: src/context/AuthContext.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        }
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    initAuth();
+  }, []);
+
+  const login = async (username: string, password: string) => {
```
Comment (sudip-sth) [inline src/context/AuthContext.tsx]: Its not necessary to put api calls in context

File: src/pages/LoginPage.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const navigate = useNavigate();
+
+  const handleSubmit = async (e: React.FormEvent) => {
+    e.preventDefault();
+    try {
+      await login(username, password);
+      navigate("/");
+    } catch {
```
Comment (sudip-sth) [inline src/pages/LoginPage.tsx:25]: add success message on successful login like toast

File: src/pages/LoginPage.tsx [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+  return (
+    <Container maxWidth="xs">
+      <Typography variant="h4" align="center" gutterBottom>
+        Login
+      </Typography>
+      <form onSubmit={handleSubmit}>
+        <TextField
+          label="Username"
+          variant="outlined"
```
Comment (sudip-sth) [inline src/pages/LoginPage.tsx:38]: Always add name and type props for formfields

File: src/services/authService.ts [hunk 5]
```diff
@@ -0,0 +1,81 @@
+import axios from "axios";
+
+const API_URL = "https://dummyjson.com/auth";
```
Comment (sudip-sth) [inline src/services/authService.ts]: put this variable in constants+

**Review by sudip-sth — CHANGES_REQUESTED** [review]

**Review by Pradip-Singh-X — COMMENTED** [review]
implemented the const variables for localStorage keys as it must be accessed across multiple components.

**Comment by Pradip-Singh-X** [comment]
separated API calls from context file, implemented environment variable, and used const variables across service file for localstorage

**Comment by Pradip-Singh-X** [comment]
added necessary props in the input fields (ie. name, type)

**Comment by Pradip-Singh-X** [comment]
implemented react-toastify for success or failure toast notifications

File: src/services/authService.ts [hunk 6]
```diff
@@ -0,0 +1,82 @@
+import axios from "axios";
+import { AUTH_TOKEN_KEY, REFRESH_TOKEN_KEY } from "../constants/constants";
+
+const API_URL = import.meta.env.VITE_SERVER_BASE_URL;
```
Comment (sudip-sth) [inline src/services/authService.ts]: keep this in constant

**Review by sudip-sth — CHANGES_REQUESTED** [review]

**Comment by Pradip-Singh-X** [comment]
**login credentials:**

username: emilys
password: emilyspass

**Pull request merged**